src/ResourceManager/Sql/Commands.Sql/ManagedInstance/Cmdlet/NewAzureSqlManagedInstance.cs

```diff
@@ -0,0 +1,198 @@
+    {
+        /// <summary>
+        /// Gets or sets the name of the managed instance name.
+        /// </summary>
+        [Parameter(Mandatory = true,
+            HelpMessage = "SQL Database Managed Instance name.")]
+        [Alias("Name")]
+        [ValidateNotNullOrEmpty]
+        public string ManagedInstanceName { get; set; }
+
+        /// <summary>
+        /// Gets or sets the admin credential of the Managed instance
+        /// </summary>
+        [Parameter(Mandatory = false, HelpMessage = "The SQL authentication credential of the Maanged Instance.")]
+        [ValidateNotNull]
+        public PSCredential AdministratorCredential { get; set; }
+
+        /// <summary>
+        /// The location in which to create the Managed instance
+        /// </summary>
+        [Parameter(Mandatory = true,
+            HelpMessage = "The location in which to create the Managed instance")]
+        [LocationCompleter("Microsoft.Sql/managedInstances")]
+        [ValidateNotNullOrEmpty]
+        public string Location { get; set; }
+
+        /// <summary>
+        /// Gets or sets the Managed instance Subnet Id
+        /// </summary>
+        [Parameter(Mandatory = true,
+            HelpMessage = "The Subnet Id to use for Managed instance creation")]
+        [ValidateNotNullOrEmpty]
+        public string SubnetId { get; set; }
+
+        /// <summary>
+        /// Gets or sets the Managed instance License Type
+        /// </summary>
+        [Parameter(Mandatory = true,
+            HelpMessage = "Determines which License Type of Sql Azure Managed instance to use")]
+        [ValidateSet(Constants.LicenseTypeBasePrice, Constants.LicenseTypeLicenseIncluded, IgnoreCase = false)]
+        public string LicenseType { get; set; }
+
+        /// <summary>
+        /// Gets or sets the Storage Size in GB for Managed instance
+        /// </summary>
+        [Parameter(Mandatory = true,
+            HelpMessage = "Determines how much Storage size to associate with Managed instance")]
+        [ValidateNotNullOrEmpty]
+        public int StorageSizeInGB { get; set; }
+
+        /// <summary>
+        /// Gets or sets the VCores for Managed instance
+        /// </summary>
+        [Parameter(Mandatory = true,
+            HelpMessage = "Determines how much VCores to associate with Managed instance")]
+        [ValidateNotNullOrEmpty]
+        public int Vcore { get; set; }
+
+        /// <summary>
+        /// Gets or sets the managed instance version
+        /// </summary>
+        [Parameter(Mandatory = true,
+            HelpMessage = "The compute generation for the Sql Azure Managed Instance. e.g. 'GP_Gen4', 'BC_Gen4'.")]
+        [ValidateNotNullOrEmpty]
+        [ValidateSet(Constants.GeneralPurposeGen4, Constants.GeneralPurposeGen5, Constants.BusinessCriticalGen4, Constants.BusinessCriticalGen5, IgnoreCase = false)]
+        public string RequestedSkuName { get; set; }
```

Comment:
Remove `-Requested`. I am guessing you based this on `New-Db -RequestedServiceObjectiveName`, which IMO is unnecessarily long. It is obviously requested if I am requesting it as a parameter :)

Also let's discuss syncing up the design of this with DB sku params.

Reply:
Use PSArgumentValidator to allow user to type in custom values for forward compatibility

Reply:
Use PSArgumentCompleter (as here: https://github.com/Azure/azure-powershell/blob/ffb67e35469d82760e7dcae4feaece9bc9605dee/src/ResourceManager/Compute/Commands.Compute/Generated/Image/Config/AddAzureRmImageDataDiskCommand.cs), which does what Jared suggests

Reply:
Done